src/components/views/elements/PowerSelector.tsx

```diff
@@ -15,40 +15,52 @@ limitations under the License.
 */
 
 import React from 'react';
-import PropTypes from 'prop-types';
 import * as Roles from '../../../Roles';
 import { _t } from '../../../languageHandler';
 import Field from "./Field";
 import { Key } from "../../../Keyboard";
 import { replaceableComponent } from "../../../utils/replaceableComponent";
 
-@replaceableComponent("views.elements.PowerSelector")
-export default class PowerSelector extends React.Component {
-    static propTypes = {
-        value: PropTypes.number.isRequired,
-        // The maximum value that can be set with the power selector
-        maxValue: PropTypes.number.isRequired,
+const CUSTOM_VALUE = "SELECT_VALUE_CUSTOM";
 
-        // Default user power level for the room
-        usersDefault: PropTypes.number.isRequired,
+interface IProps {
+    value: number;
+    // The maximum value that can be set with the power selector
+    maxValue: number;
 
-        // should the user be able to change the value? false by default.
-        disabled: PropTypes.bool,
-        onChange: PropTypes.func,
+    // Default user power level for the room
+    usersDefault: number;
 
-        // Optional key to pass as the second argument to `onChange`
-        powerLevelKey: PropTypes.string,
+    // should the user be able to change the value? false by default.
+    disabled?: boolean;
+    onChange?: (value: string, powerLevelKey: string) => void;
```

Comment:
`onChange` should probably take a number instead

Reply:
```
src/components/views/right_panel/UserInfo.tsx(1119,17): error TS2322: Type '(powerLevelStr: string) => Promise<void>' is not assignable to type '(value: number, powerLevelKey: string) => void'.
  Types of parameters 'powerLevelStr' and 'value' are incompatible.
    Type 'number' is not assignable to type 'string'.
src/components/views/settings/tabs/room/RolesRoomSettingsTab.tsx(311,29): error TS2322: Type '(value: string, powerLevelKey: string) => void' is not assignable to type '(value: number, powerLevelKey: string) => void'.
  Types of parameters 'value' and 'value' are incompatible.
    Type 'number' is not assignable to type 'string'.
src/components/views/settings/tabs/room/RolesRoomSettingsTab.tsx(322,29): error TS2322: Type '(value: string, powerLevelKey: string) => void' is not assignable to type '(value: number, powerLevelKey: string) => void'.
src/components/views/settings/tabs/room/RolesRoomSettingsTab.tsx(403,21): error TS2322: Type '(inputValue: string, powerLevelKey: string) => void' is not assignable to type '(value: number, powerLevelKey: string) => void'.
  Types of parameters 'inputValue' and 'value' are incompatible.
    Type 'number' is not assignable to type 'string'.
src/components/views/settings/tabs/room/RolesRoomSettingsTab.tsx(432,25): error TS2322: Type '(inputValue: string, powerLevelKey: string) => void' is not assignable to type '(value: number, powerLevelKey: string) => void'.
```

Components seem to be expecting a string

Reply:
I'd argue those components are wrong.

Reply:
I wasn't thinking yesterday, sorry. It should actually take both - number in case it's custom and a string if it's not custom

Reply:
Oh, you're actually right :facepalm: